src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/FileUpload/FileUploadWithTag/FileUploadWithTagComponent.tsx

```diff
@@ -0,0 +1,257 @@
+import { getLanguageFromKey } from 'altinn-shared/utils';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import { isMobile } from 'react-device-detect';
+import { IAttachment } from '../../../../shared/resources/attachments';
+import AttachmentDispatcher from '../../../../shared/resources/attachments/attachmentActions';
+import { IRuntimeState , IComponentValidations } from '../../../../types';
+import { renderValidationMessagesForComponent } from '../../../../utils/render';
+import { FormLayoutActions } from 'src/features/form/layout/formLayoutSlice';
+import { v4 as uuidv4 } from 'uuid';
+import { getFileUploadWithTagComponentValidations, isAttachmentError, isNotAttachmentError, parseFileUploadComponentWithTagValidationObject } from 'src/utils/formComponentUtils';
+import { baseStyle, activeStyle, rejectStyle, validationErrorStyle } from 'src/styles/Dropzone';
+import { renderAttachmentsCounter, renderFileUploadContent } from '../shared/render';
+import { FileList } from './FileListComponent';
+
+export interface IFileUploadWithTagProps {
+  hasCustomFileEndings?: boolean;
+  id: string;
+  isValid?: boolean;
+  componentValidations?: IComponentValidations;
+  language: any;
+  optionsId: string;
+  maxFileSizeInMB: number;
+  maxNumberOfAttachments: number;
+  minNumberOfAttachments: number;
+  readOnly: boolean;
+  validFileEndings?: string;
+  getTextResource: (key: string) => string;
+  getTextResourceAsString: (key: string) => string;
+  textResourceBindings: any;
+}
+
+export const bytesInOneMB = 1048576;
+export const emptyArray = [];
+
+export function FileUploadWithTagComponent(props: IFileUploadWithTagProps): JSX.Element {
+  const dataDispatch = useDispatch();
+  const [validations, setValidations] = React.useState<Array<{ id: string, message: string }>>([]);
+  const mobileView = useMediaQuery('(max-width:992px)'); // breakpoint on altinn-modal
+  const options = useSelector((state: IRuntimeState) => state.optionState.options[props.optionsId]);
+  // eslint-disable-next-line max-len
+  const editIndex = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.fileUploadersWithTag[props.id]?.editIndex ?? -1);
+  // eslint-disable-next-line max-len
+  const chosenOptions = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.fileUploadersWithTag[props.id]?.chosenOptions ?? {});
+
+  const attachments: IAttachment[] = useSelector(
+    (state: IRuntimeState) => state.attachments.attachments[props.id] || emptyArray,
+  );
+
+  const setValidationsFromArray = (validationArray: string[]) => {
+    setValidations(
+      validations.concat(parseFileUploadComponentWithTagValidationObject(validationArray)),
+    );
+  };
+
+  const setEditIndex = (index: number) => {
+    dataDispatch(FormLayoutActions.updateFileUploaderWithTagEditIndex({
+      uploader: props.id, index
+    }));
+  };
+
+  const onClick = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
```

Comment:
Renamed to 'handleClick'. Unsure about the other 'onSomething' functions in this file, as they are passed as props as you mentioned. Is this acceptable?

Reply:
Most of these should be called `handleSomething`. Lets take the `onClickSave` as an example.

You are "handling" the save event in this function, so it should be called f.ex `handleSave`. This callback is passed to the `FileList` component, and the `FileList` component should accept a prop with `on*` name (f.ex `onSave`). This is where the event is triggered. (The `click` part of the name may not be important, you could be saving based on some other type of event, f.ex keyboard press).

```jsx
// in FileUploadWithTagComponent.tsx
const FileUploadWithTagComponent = () => {
  const handleSave = (attachment) {
    // handle the save
  }

  //...
  <FileList
    onSave={handleSave}
  />
}

// in FileListComponent.tsx
const FileList = ({
  onSave,
}) => {
//...
  <AltinnButton
    onClickFunction={() => props.onSave(props.attachment)}
  />
}

// ----------------------------------------
// Alternative in FileListComponent.tsx, to use local handler instead of inline anonoymous function to handle the event
const FileList = ({
  onSave,
}) => {

  const handleSaveClick = () => {
    props.onSave(props.attachment)
  }
//...
  <AltinnButton
    onClickFunction={handleSaveClick}
  />
}

```


So in general, when a component accepts an event handler, the prop name should be `onSomething`. When a function handles said event, it should be called `handleSomething`.

Did that make sense?

Reply:
Yes, it made perfect sense. Thank you.
Made the changes.